lang/parser/src/grammar/cst.lalrpop

```diff
@@ -5,33 +5,73 @@ use num_bigint::BigUint;
 use crate::cst::exp::*;
 use crate::cst::decls::*;
 use crate::cst::ident::*;
+use crate::lexer::{Token, LexicalError};
 
 use super::util::span;
 
 #[LALR]
 grammar;
 
 // Tokens
-match {
+extern {
+  type Location = usize;
+  type Error = LexicalError;
+
+  enum Token {
+    // Keywords
+    //
+    //
+    "data" => Token::Data,
+    "codata" => Token::Codata,
+    "def" => Token::Def,
+    "codef" => Token::Codef,
+    "let" => Token::Let,
+    "match" => Token::Match,
+    "as" => Token::As,
+    "comatch" => Token::Comatch,
+    "absurd" => Token::Absurd,
+    "..absurd" => Token::DotsAbsurd,
+    "Type" => Token::Type,
+
+    // Parens, Braces and Brackets
+    //
+    //
+    "(" => Token::LParen,
+    ")" => Token::RParen,
+    "{" => Token::LBrace,
+    "}" => Token::RBrace,
+    "[" => Token::LBracket,
+    "]" => Token::RBracket,
+
     // Symbols
-    "(", ")", "{", "}", "[", "]", ";", ":=", "=>", ",", ":", ".", "?",
-    "->", "\\", "#",
+    //
+    //
+    ";" => Token::Semicolon,
+    ":=" => Token::ColonEq,
+    "=>" => Token::DoubleRightArrow,
+    "," => Token::Comma,
+    ":" => Token::Colon,
+    "." => Token::Dot,
+    "?" => Token::QuestionMark,
+    "->" => Token::RightArrow,
+    "\\" => Token::Backslash,
+    "#" => Token::Hash,
+    "_" => Token::Underscore,
+
     // Names
-    "_",
-    r"[a-zα-ω][a-zα-ωA-ZΑ-Ω0-9_]*['⁺⁻₀₁₂₃₄₅₆₇₈₉₊₋]*",
-    r"[A-ZΑ-Ω𝔹ℕ𝕍∃∀×][a-zα-ωA-ZΑ-Ω0-9_]*['⁺⁻₀₁₂₃₄₅₆₇₈₉₊₋]*",
+    //
+    //
+    "UpperCaseName" => Token::UpperCaseName(<String>),
+    "LowerCaseName" => Token::LowerCaseName(<String>),
```

Comment:
It would also be cool to get rid of this distinction, eventually.

Reply:
I think there is only one thing standing in the way: a parser ambiguity in the declaration of self parameters for destructors, right? We probably should open an issue about this and discuss how we want to handle this parsing ambiguity.